src/window.rs

```diff
@@ -357,6 +372,23 @@ impl Window {
     pub fn request_redraw(&self) {
         self.window.request_redraw()
     }
+
+    /// Sets the event handler for a window and returns any previously set event handler.
+    ///
+    /// If an event handler is set, the handler is called for events with `window_id == window.id()`.
+    ///
+    /// This function is mainly intended for embedded windows with an external parent window such as
+    /// when using `WindowBuilderExtWindows.with_parent_window` where external code controls the main
+    /// event loop. However, it can also be used with `EventLoop.run` for per-window event handling.
+    /// In such cases, the event handler is called before the main event handler passed to
+    /// `EventLoop.run` is called.
+    #[inline]
+    pub fn set_event_handler<T: 'static>(
+        &self,
+        _event_handler: Option<Box<dyn FnMut(Event<T>) -> bool>>,
```

Comment:
Presumably this callback should also be static? See https://github.com/rust-windowing/winit/blob/master/src/event_loop.rs#L138

Won't that interfere with `run`?

What is this function mean to be returning?

Reply:
I'm not sure about the lifetime. I doubt it should be `'static`, as each Window could have its own handler and you don't know a priori how many windows you will have.

Why would it interfere with `run`?

My initial thought with the `bool` return value was that events that are handled by an `event_handler` would no longer be passed to the event handler passed to `run`, but it doesn't necessarily have to be implemented that way. In the embedded case, the `run` handler can't really exist anyways - at least not as is.